tests/ast/data/Dataframe.to_snowpark_pandas.test

```diff
@@ -0,0 +1,189 @@
+## TEST CASE
+
+df = session.table(tables.table1)
+snowpark_pandas_df = df.to_snowpark_pandas()
+
+snowpark_pandas_df = df.to_snowpark_pandas(index_col="A")
+
+snowpark_pandas_df = df.to_snowpark_pandas(columns=["B", "A"])
+
+snowpark_pandas_df = df.to_snowpark_pandas(index_col=["A"], columns=["C", "B"])
+
+# Perform a pandas operation on the result to ensure nothing breaks.
```

Comment:
So I think this operation should then produce with AST enabled the internal Snowpark calls, or should it not? Just trying to understand the logic here. Perhaps you'll need to add `snowpark_pandas_df.groupby("A").to_numpy()` or so to trigger an eval... Yet our test logic should flush the last batch no matter what.

Reply:
It won't produce an AST because I don't think we added any pandas AST logic

Reply:
To avoid scope creep, I think it's a good idea to test the pandas AST logic in a different PR. This PR should only focus on whether the `to_snowpark_pandas` protobuf is generated correctly. Here's the ticket for it: https://snowflakecomputing.atlassian.net/browse/SNOW-1849281

Testing the pandas logic with local testing/nop testing is very janky so it will have to be in a separate test file.